SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebuggerEditor.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Utils/SpatialDebuggerEditor.h"
+
+#include "EngineClasses/SpatialWorldSettings.h"
+#include "Interop/Connection/SpatialWorkerConnection.h"
+#include "LoadBalancing/GridBasedLBStrategy.h"
+#include "LoadBalancing/LayeredLBStrategy.h"
+#include "SpatialCommonTypes.h"
+#include "Utils/InspectionColors.h"
+
+#include "Debug/DebugDrawService.h"
+#include "Engine/World.h"
+#include "GameFramework/WorldSettings.h"
+#include "Modules/ModuleManager.h"
+
+using namespace SpatialGDK;
+
+ASpatialDebuggerEditor::ASpatialDebuggerEditor(const FObjectInitializer& ObjectInitializer)
+	: Super(ObjectInitializer)
+{
+	PrimaryActorTick.bCanEverTick = false;
+	bReplicates = false;
+}
+
+void ASpatialDebuggerEditor::Destroyed()
+{
+	DestroyWorkerRegions();
+}
+
+void ASpatialDebuggerEditor::ToggleWorkerRegionVisibility(bool bEnabled)
+{
+	bShowWorkerRegions = bEnabled;
+	RefreshWorkerRegions();
+}
+
+void ASpatialDebuggerEditor::RefreshWorkerRegions()
+{
+
+	DestroyWorkerRegions();
+
+	if (bShowWorkerRegions && AllowWorkerBoundaries())
+	{
+		InitialiseWorkerRegions();
+		CreateWorkerRegions();
+	}
+#if WITH_EDITOR
+	if (GEditor != nullptr && GEditor->GetActiveViewport() != nullptr)
+	{
+		// Redraw editor window to show changes
+		GEditor->GetActiveViewport()->Invalidate();
+	}
+#endif
+}
+
+bool ASpatialDebuggerEditor::AllowWorkerBoundaries() const
+{
+	// Check if multi worker is enabled.
+	UWorld* World = GetWorld();
+
+	if (World == nullptr)
+	{
+		return false;
+	}
```

Comment:
Can we make this a `check(...)`? Or is there a valid case for calling this function with a nullptr World?

Reply:
Sure, done.